internal/pkg/aws/cloudformation/stackset/stackset.go

```diff
@@ -112,27 +112,66 @@ func (ss *StackSet) UpdateAndWait(name, template string, opts ...CreateOrUpdateO
 	return ss.WaitForOperation(name, id)
 }
 
-// DeleteAllInstances removes all stack instances from a stack set and returns the operation ID.
-// If the stack set does not exist, then return [ErrStackSetNotFound].
-// If the stack set does not have any instances, then return [ErrStackSetInstancesNotFound].
-// Both errors should satisfy [IsEmptyStackSetErr], otherwise it's an unexpected error.
-func (ss *StackSet) DeleteAllInstances(name string) (string, error) {
+func (ss *StackSet) getStackSetUniqueAccountsAndRegions(name string) ([]InstanceSummary, error) {
 	summaries, err := ss.InstanceSummaries(name)
 	if err != nil {
 		// If the stack set doesn't exist - just move on.
 		if isNotFoundStackSet(errors.Unwrap(err)) {
-			return "", &ErrStackSetNotFound{
+			return nil, &ErrStackSetNotFound{
 				name: name,
 			}
 		}
-		return "", err
+		return nil, err
 	}
 
 	if len(summaries) == 0 {
-		return "", &ErrStackSetInstancesNotFound{
+		return nil, &ErrStackSetInstancesNotFound{
 			name: name,
 		}
 	}
+	return summaries, nil
+}
+
+func (ss *StackSet) DeleteInstance(name, account, region string) (string, error) {
+	summaries, err := ss.getStackSetUniqueAccountsAndRegions(name)
+	if err != nil {
+		return "", err
+	}
+	accountAndRegionPresentInInstances := false
+	for _, summary := range summaries {
+		if summary.Region == region && summary.Account == account {
+			accountAndRegionPresentInInstances = true
+			break
+		}
+	}
+
+	// No-op for idempotency if the account and region is not found in the deployed instances.
+	if !accountAndRegionPresentInInstances {
+		return "", nil
+	}
+
+	out, err := ss.client.DeleteStackInstances(&cloudformation.DeleteStackInstancesInput{
```

Comment:
What would happen if we just call `DeleteStackInstances` without first checking the summaries?

Reply:
hmm it's actually idempotent 🤔 Going to just remove the check and put the stackset summary logic in the CLI package. This is an unfortunately leaky abstraction here, since we don't have copilot concepts down at the level of stackset operations.